File: library_common/src/main/java/com/xxjy/common/util/LoginHelper.java
//package com.xxjy.common.util;
//
//import android.app.Activity;
//import android.content.Context;
//import android.content.Intent;
//
//import com.blankj.utilcode.util.SPUtils;
//import com.xxjy.jyyh.constants.SPConstants;
//import com.xxjy.jyyh.ui.login.LoginActivity;
//import com.xxjy.jyyh.utils.umengmanager.UMengManager;
//
///**
// * @author power
// * @date 12/4/20 3:31 PM
// * @project RunElephant
// * @description:
// */
//public class LoginHelper {
//    public static CallBack callBack;
//
//    public static void login(Context context, CallBack loginCallBack) {
//        if (!SPUtils.getInstance().getBoolean(SPConstants.LOGIN_STATUS)) {
//            callBack = loginCallBack;
//            Intent intent = new Intent(context, LoginActivity.class);
//            context.startActivity(intent);
//        } else {
//            if (loginCallBack != null) {
//                loginCallBack.onLogin();
//            }
//        }
//    }
//
//    public static void loginOut(Activity activity) {
//        SPUtils.getInstance().put(SPConstants.LOGIN_STATUS,false);
//        SPUtils.getInstance().put(SPConstants.MOBILE,"");
//        SPUtils.getInstance().put(SPConstants.APP_TOKEN,"");
//        SPUtils.getInstance().put(SPConstants.USER_TYPE,-1);
//        SPUtils.getInstance().put(SPConstants.OPEN_ID,"");
//        SPUtils.getInstance().put(SPConstants.NEW_USER_RED_PACKET,0l);
//        UMengManager.onProfileSignOff();
//        if (activity != null){
//            activity.finish();
//        }
//    }
//
//    public interface CallBack {
//        void onLogin();
//    }
//}

File: library_common/src/main/java/com/xxjy/common/util/LoginHelper.kt
package com.xxjy.common.util

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import com.xxjy.common.constants.UserConstants
import com.xxjy.common.router.ARouterManager
import com.xxjy.common.router.RouteConstants
import com.xxjy.umeng.UMengManager

/**
 * @author power
 * @date 12/4/20 3:31 PM
 * @project RunElephant
 * @description:
 */
object LoginHelper {
    var callBack: CallBack? = null
    fun login(loginCallBack: CallBack?) {
        if (!UserConstants.login_status) {
            callBack = loginCallBack
         ARouterManager.navigation(RouteConstants.Personal.A_LOGIN).navigation()
        } else {
            loginCallBack?.onLogin()
        }
    }

    fun loginOut(activity: AppCompatActivity?) {
        UserConstants.login_status=false
        UserConstants.mobile=""
        UserConstants.token=""
        UserConstants.user_type=-1
        UserConstants.openId=""
        UserConstants.new_user_red_packet=0L



        UMengManager.onProfileSignOff()
        activity?.finish()
    }

    interface CallBack {
        fun onLogin()
    }
}